lib/rubocop/cop/style/endless_range.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # This cop checks that arrays are sliced with endless ranges instead of
+      # [start..-1] on Ruby 2.6+
+      #
+      # @example
+      #   # bad
+      #   items[1..-1]
+      #
+      #   # good
+      #   items[1..]
+      class EndlessRange < Cop
+        extend TargetRubyVersion
+
+        minimum_target_ruby_version 2.6
+
+        MSG = 'Prefer endless ranges (start..) for slicing '\
+              'till the end of array.'
+
+        def_node_matcher :range_till_minus_one?, '(irange (int _) (int -1))'
+
+        def on_send(node)
+          return unless node.method?(:[]) && node.arguments.count == 1
```

Comment:
I think you could handle all of these checks with a single `def_node_matcher`. It's been a little while since I wrote one, I think it would look like `(send _ :[] (irange (int _) (int -1)))`

Reply:
Of course I can, but I believe it is better this way for concern separation.

"Works only for method `#[]` with a singular argument" is an arbitrary decision, which can be changed in future and/or depend on settings (for ex., one may also expect `#slice` to be checked, or multi-argument `#[]` when workin with matrices). But "range ending in `-1`" is what the cop is about. So I believe it is better to leave as two separate checks.